xbmc/addons/binary-addons/AddonInstanceHandler.cpp

```diff
@@ -59,8 +94,7 @@ bool IAddonInstanceHandler::CreateInstance(KODI_HANDLE instance)
 
 void IAddonInstanceHandler::DestroyInstance()
 {
-  if (m_addon)
-    m_addon->DestroyInstance(m_instanceId);
+  m_addon->DestroyInstance(m_instanceId);
```

Comment:
is it guaranteed that m_addon is always set here? In all (many?) other functions you take care to check before access. why not here?